hack/create-cluster.sh

```diff
@@ -47,13 +47,35 @@ if [ -f "${KUBECONFIG}" ];then
   fi
 fi
 
-kind create cluster --name "${CLUSTER_NAME}" --kubeconfig="${KUBECONFIG}" --wait=120s
+util::get_macos_ipaddress # Adapt for macOS
+
+# create a cluster to deploy karmada control plane components.
+if [[ -n "${MAC_NIC_IPADDRESS}" ]]; then
+    CONF=`mktemp`
+    PUBLIC_PORT=$(((RANDOM%(8000-7000+1))/1*1+7000))
+    cp artifacts/kindClusterConfig/member-example.yaml "${CONF}"
+    sed -i '' "s/MAC_NIC_IPADDRESS/${MAC_NIC_IPADDRESS}/g" "${CONF}"
+    sed -i '' "s/PUBLIC_PORT/${PUBLIC_PORT}/g" "${CONF}"
```

Comment:
`sed -i'' -e` will work the most versions of macOS and Linux
```
sed -i'' -e "s/MAC_NIC_IPADDRESS/${MAC_NIC_IPADDRESS}/g" "${CONF}"
sed -i'' -e "s/PUBLIC_PORT/${PUBLIC_PORT}/g" "${CONF}"
```

Reply:
These codes will only run in macos